src/dom/Decimal.ts

```diff
@@ -27,13 +27,44 @@ export class Decimal extends Value(
 
   /**
    * Constructor.
-   * @param value         The numeric value to represent as a decimal.
+   * @param value         The Ion decimal value to represent as a decimal.
    * @param annotations   An optional array of strings to associate with `value`.
    */
-  constructor(value: ion.Decimal, annotations: string[] = []) {
-    super(value.numberValue());
-    this._decimalValue = value;
-    this._numberValue = value.numberValue();
+  constructor(value: ion.Decimal, annotations?: string[]);
+
+  /**
+   * Constructor.
+   * @param value         The string value to represent as a decimal.
+   * @param annotations   An optional array of strings to associate with `value`.
+   */
+  constructor(value: string, annotations?: string[]);
+
+  /**
+   * Constructor.
+   * @param value         The number value to represent as a decimal.
+   * @param annotations   An optional array of strings to associate with `value`.
+   */
+  constructor(value: number, annotations?: string[]);
+
+  // This is the unified implementation of the above signatures and is not visible to users.
+  constructor(
+    value: ion.Decimal | string | number,
+    annotations: string[] = []
+  ) {
+    if (typeof value === "string") {
+      super(Number(value));
+      this._decimalValue = new ion.Decimal(value);
+      this._numberValue = Number(value);
+    } else if (value instanceof ion.Decimal) {
+      super(value.numberValue());
+      this._decimalValue = value;
+      this._numberValue = value.numberValue();
+    } else {
+      // if value is a number type
+      super(value);
```

Comment:
In a TypeScript program, this is correct. If this is run in JavaScript, I believe `value` could be another non-`number` type.
```suggestion
    } else if (typeof value === "number") {
      // if value is a number type
      super(value);
```
with a corresponding `else` that throws for any other type would be safer.

Reply:
Yes, that makes sense. I will change it.